### ISSUE TITLE
Component fails axe tests due to role="presentation" on button element

### ISSUE THREAD

**Issue by web-padawan** [vaadin/vaadin-button#144]
Mentioned in #96 but is a separate problem, so I decided to extract from there.

```
Rule: aria-allowed-role
Impact: minor
ARIA role must be appropriate for the element (https://dequeuniversity.com/rules/axe/3.4/aria-allowed-role?application=axeAPI)

Issue target: vaadin-drawer-toggle,button
Context: <button type="button" role="presentation" tabindex="0"></button>
Fix any of the following:
  ARIA role presentation  is not allowed for given element
```

We tested [Material Web Components demos](https://mwc-demos.glitch.me/demos/) and they don't use the `role="button"` on the host element. Need to test if that would work for us too.

**Comment by web-padawan** [comment]
Here is the current list of [allowed elements](https://github.com/dequelabs/axe-core/blob/ad1448088b93e7e9bb51a5f854234266c1e98c51/lib/commons/aria/index.js#L1483-L1523) that can use `presentation` role according to `axe`.

**Comment by web-padawan** [comment]
See https://www.w3.org/TR/using-aria/#4thrule

**Comment by anezthes** [comment]
I tested the following HTML structure with great success in Chrome and Safari using VoiceOver:
```
<button>
  <span part="prefix"><slot name="prefix"></slot></span>
  <span><slot></slot></span>
  <span part="suffix"><slot name="suffix"></slot></span>
</button>
```

**Comment by rolfsmeds** [comment]
This seems to cause some screen readers to announce button labels twice or not at all.

**Comment by rolfsmeds** [comment]
Closing in favor of https://github.com/vaadin/vaadin-button/issues/96
